src/protocols/CHIPProtocols.h

```diff
@@ -0,0 +1,66 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines constant enumerations for all public (or
+ *      common) protocols.
+ *
+ */
+
+#ifndef CHIP_PROTOCOLS_H_
+#define CHIP_PROTOCOLS_H_
+
+#include <core/CHIPVendorIdentifiers.hpp>
+
+/**
+ *   @namespace chip::Protocols
+ *
+ *   @brief
+ *     This namespace includes all interfaces within CHIP for CHIP protocols.
+ */
+
+namespace chip {
+namespace Protocols {
+
+//
+// CHIP Protocol Ids (32-bits max)
+//
+
+enum CHIPProtocolId
```

Comment:
no need to prefix this with CHIP, it's already in the CHIP namespace, right?

Reply:
also: I believe the message layer may already have a type in place for this

Reply:
This is port from /lib/protocols since the legacy wml based message/protocol stack need to be discarded. The references of enum CHIPProtocolId are throughout of current CHIP stack and beyond this PR. We need to address this issue separately.

Reply:
will you please file an issue and reference here?

Reply:
Here we go. https://github.com/project-chip/connectedhomeip/issues/3479